### DISCUSSION TITLE
Datasource REST API

### DISCUSSION THREAD

**Discussion by Abd-Atif** [Budibase/budibase#9731] [Help] [answered]
Hi Folks,
I'm new to Budibase, so I'm adding a retry button in the page that is triggering an automation which is calling an API (method = POST) but issue is how to pass request (body) to the API via bindings or any other option?
Kindly elaborate the ways to get through it, thanks

here are some screenshots for better understanding:
![image](https://user-images.githubusercontent.com/94058099/219853547-637f1d12-760f-4756-a8b6-aa6aa9c676d5.png)

![image](https://user-images.githubusercontent.com/94058099/219853557-ca3419d1-c416-41cc-9c76-731d15526926.png)

![image](https://user-images.githubusercontent.com/94058099/219853618-0d2fcbb7-b5df-4f28-be0d-24b3029c0a10.png)

**Comment by Abd-Atif** [comment]
bump

**Reply by Abd-Atif** [reply]
@melohagan

**Reply by Abd-Atif** [reply]
@melohagan 
@skorphil
@andz-bb 
kindly help me guys

**Comment by Abd-Atif** [comment, marked as answer]
now Im able to pass body in the POST request but its adding an extra variable "bodyPart" to the actual body.
Is this approach is fine? or is there any issues?
please guide

Lemme add screenshots for better understanding.

![image](https://user-images.githubusercontent.com/94058099/219954933-c0570b47-331e-421b-8953-bc097324d0ff.png)

![image](https://user-images.githubusercontent.com/94058099/219954952-f301b291-7e44-49e3-a886-bf73abac5866.png)

![image](https://user-images.githubusercontent.com/94058099/219954964-c40fda63-9ffc-48dc-b8ae-48ccd26f8e85.png)

![image](https://user-images.githubusercontent.com/94058099/219954980-0f258bf2-3b1d-4471-9d96-2c25f90b56a2.png)

![image](https://user-images.githubusercontent.com/94058099/219954989-e7abe4aa-7de2-471f-958f-e26f090e080e.png)
 
 _ALSO, I want my request **body** should be like mentioned below:_
 
![image](https://user-images.githubusercontent.com/94058099/219955187-7331539c-618b-4516-b33c-48697b2c762c.png)


kindly give solutions & suggestions/ideas, **THANKS**

**Reply by melohagan** [reply]
Hey @Abd-Atif 

Usually instead of passing in the entire payload body as single binding I would do each field separately, e.g.

```js
{
  "templateId": {{ templateId }},
  "language": {{ language }},
  etc...
}
```

**Reply by Abd-Atif** [reply]
Thanks for you suggestion but in my case a column field is holding whole request payload.
Also how to process 2 different fields as a common body (request payload) in POST request?

**Reply by Abd-Atif** [reply]
@melohagan how to pass two different values as a body in this POST request
